setup.py

```diff
@@ -37,19 +38,51 @@ def process_ignore_tags(buffer):
         return None
 
 
+# Note: why do we bother including a 'scapy/VERSION' file and doing our
+# own versioning stuff, instead of using more standard methods?
+# Because it's all garbage.
+
+# If you remain fully standard, there's no way
+# of adding the version dynamically, even less when using archives
+# (currently, we're able to add the version anytime someone exports Scapy
+# on github).
+
+# If you use setuptools_scm, you'll be able to have the git tag set into
+# the wheel (therefore the metadata), that you can then retrieve using
+# importlib.metadata, BUT it breaks sdist (source packages), as those
+# don't include metadata.
+
+
+def _build_version(path):
+    """
+    This adds the scapy/VERSION file when creating a sdist and a wheel
+    """
+    fn = os.path.join(path, 'scapy', 'VERSION')
+    with open(fn, 'w') as f:
+        f.write(__import__('scapy').VERSION)
+
+
 class SDist(sdist):
     """
     Modified sdist to create scapy/VERSION file
     """
     def make_release_tree(self, base_dir, *args, **kwargs):
         super(SDist, self).make_release_tree(base_dir, *args, **kwargs)
         # ensure there's a scapy/VERSION file
-        fn = os.path.join(base_dir, 'scapy', 'VERSION')
-        with open(fn, 'w') as f:
-            f.write(__import__('scapy').VERSION)
+        _build_version(base_dir)
+
+
+class BuildPy(build_py):
+    """
+    Modified build_py to create scapy/VERSION file
+    """
+    def build_package_data(self):
+        super(BuildPy, self).build_package_data()
+        # ensure there's a scapy/VERSION file
+        _build_version(self.build_lib)
```

Comment:
I think now that `scapy/VERSION` is added to the sdist and wheel automatically it can be removed from `MANIFEST.in` to prevent `setuptools` from complaining about it:
```sh
warning: no files found matching 'scapy/VERSION'
```


FWIW there seems to be another warning when the wheel is generated from the sdist:
```sh
warning: no previously-included files matching '*' found under directory 'test'
```
but it can be reproduced with the master branch as well so it's probably a separate issue.

Reply:
> FWIW there seems to be another warning when the wheel is generated from the sdist:

It was introduced in 6ae0e87fba24e9e61eaf687d5374a5dbd330088d as far as I can see.

Reply:
The two comments appear to be correct !